src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1932,24 +1932,34 @@ void PerformKeepAlive(object? state)
 
         #region Wait
 
-        public bool Wait(int timeout)
+        public async Task<bool> Wait(bool async, int timeout, CancellationToken cancellationToken = default)
         {
-            if ((timeout != 0 && timeout != -1) && IsSecure)
-                throw new NotSupportedException("Wait() with timeout isn't supported when SSL is used, see https://github.com/npgsql/npgsql/issues/1501");
+#if NET461
+            if (timeout > 0 && IsSecure)
+                throw new NotSupportedException("Wait with timeout isn't supported when SSL is used on .NET Framework. Please consider moving to .NET Core or disabling SSL.");
+
+            if (timeout > 0 && async)
+                throw new NotSupportedException("WaitAsync with timeout isn't supported when used on .NET Framework. Please consider moving to .NET Core.");
+
+            if (timeout <= 0 && cancellationToken.CanBeCanceled)
```

Comment:
Shouldn't we always through for CanBeCanceled, regardless of timeout?

Reply:
There is a case with a KeepAlive. While the CancellationToken doesn't cancel write or read from a socket, we do still have some checks for `cancellationToken.ThrowIfCancellationRequested()`.

Reply:
Hmm, OK... It's not very important, but these checks don't seem to be enough to say that WaitAsync actually supports the cancellation tokens - the really interesting case is to actually cancel a read operation, which we can't do on this TFM. So I'm leaning towards throwing immediately for any non-None cancellation token.

But I'm fine to merge either way - let me know which way you want to do it.

Reply:
Oh, I see you actually already did this. Merging.

Reply:
Well, not really - it's still working the same. We check for the timeout, and if it is less or equal to zero, then we check for the cancellation token. I don't mind if you do change it.